pkg/sideinputs/initializer/initializer.go

```diff
@@ -67,14 +74,85 @@ func (sii *sideInputsInitializer) Run(ctx context.Context) error {
 	default:
 		return fmt.Errorf("unrecognized isbsvc type %q", sii.isbSvcType)
 	}
+	// Load the required KV bucket and create a sideInputWatcher for it
+	bucketName := isbsvc.JetStreamSideInputsStoreBucket(sii.sideInputsStore)
+	sideInputWatcher, err := jetstream.NewKVJetStreamKVWatch(ctx, sii.pipelineName, bucketName, natsClient)
+	if err != nil {
+		return err
+	}
+
+	retCh := make(chan map[string][]byte, 1)
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	m := createSideInputMap(sii.sideInputs)
+	go startSideInputWatcher(ctx, sideInputWatcher, &wg, retCh, log, m)
+	wg.Wait()
```

Comment:
what is the point of this go routine, if we doing a wait, it looks like it is a synchronous call?